text/0240-app-telemetry-schema-vision-roadmap.md

```diff
@@ -0,0 +1,430 @@
+> which underpins the [dbt Semantic Layer](https://www.getdbt.com/product/semantic-layer),
+> highlights the significance of adopting a schema-first approach in data
+> modeling, especially for Generative AI-based question answering systems. Tools
+> like Observability Query Assistants (
+> e.g. [Elastic AI Assistant](https://www.elastic.co/fr/blog/introducing-elastic-ai-assistant)
+> and [Honeycomb Query Assistant](https://www.honeycomb.io/blog/introducing-query-assistant?utm_source=newswire&utm_medium=link&utm_campaign=query_assistant))
+> are likely to become increasingly prevalent and efficient in the near future,
+> thanks to the adoption of a schema-first approach.
+
+Several OTEPs will be dedicated to the precise definition of the structure and
+the format of this/these schema(s). The rules for resolving overrides
+(inheritance), external references, and conflicts will also be described in
+these OTEPs. See the Roadmap section for a comprehensive list of these OTEPs.
+
+## Resolved Telemetry Schema
+
+A Resolved Telemetry Schema is the outcome of the schema resolution process.
+This process involves taking the entire hierarchy of Telemetry Schemas and
+Semantic Convention Registries and applying a set of rules to resolve overrides
+and eliminate external references. The key design principles to be followed in
+the definition of the Resolved Telemetry Schema are:
+
+* **Self-contained**: No external references are allowed. This artifact contains
+  everything required to determine what an application or a library produces in
+  terms of telemetry.
+* **Easy to exchange**: This artifact must be easily accessible from a web
+  server via a URL. This artifact must be small and avoid the repetition of
+  definitions.
+* **Easy to parse**: A widespread and well-defined format should be preferred.
+  JSON is an example of such a format.
+* **Easy to interpret**: The internal structure of this artifact must be
+  straightforward to avoid any misinterpretation.
+* **Platform- and Language-agnostic**: This artifact must be independent of any
+  platform architectures and programming languages.
+
+The following diagram describes two main use cases for the Resolved Telemetry
+Schema. The key points to remember are: 1) both use cases result in a Resolved
+Telemetry Schema, and 2) Resolved Telemetry Schemas serve as the mechanism for
+distributing Telemetry Schemas throughout the entire ecosystem.
+
+![Use cases](./img/0240-otel-weaver-use-cases.svg)
+
+The main components of a Resolved Telemetry Schema are illustrated in the
+diagram below. The 'OTel Weaver' tool is used to create these schemas. It can
+also extend an existing schema or import a Semantic Convention Registry.
+Resolved Telemetry Schema serves as a key mechanism for interoperability,
+feeding various external tools, including SDK generators, documentation
+generators, policy enforcers, and more.
+
+![Resolved Telemetry Schema](./img/0240-otel-weaver-resolved-schema.svg)
```

Comment:
Nit: according to the diagram, a Resolved Telemetry Schema refers to _exactly one_ instrumentation library. Will this also true for the Resolved Telemetry Schema created directly from OpenTelemetry semantic conventions and provided by OpenTelemetry?

Reply:
Not exactly. The dependencies section will contains the instrumentation libraries of each dependency (see the appendix B I recently added to this OTEP). Regarding a resolved telemetry schema representing a semantic convention registry, the instrumentation library is not defined.